source/logging.c

```diff
@@ -291,6 +291,26 @@ int aws_log_level_to_string(enum aws_log_level log_level, const char **level_str
     return AWS_OP_SUCCESS;
 }
 
+int aws_thread_id_to_string(aws_thread_id thread_id, char *str, size_t length) {
+    AWS_ERROR_PRECONDITION(AWS_THREAD_ID_REPR_LEN == length);
```

Comment:
debatable: it's a little weird to make the user pass length, but there is only one correct length. Though I understand that we're concerned about overflow, so this is kind of like making the user sign a waiver each time. I could go either way...

However, I do care about this: "length" when you're talking about c-strings usually doesn't include the null terminator. In the [snprintf docs](https://en.cppreference.com/w/c/io/fprintf) they call their args `(char *buffer, size_t bufsz, ...)` maybe we should do the same?